src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyCultureInfo.cs

```diff
@@ -42,42 +42,46 @@ public override void Initialize(AnalysisContext analysisContext)
             analysisContext.RegisterCompilationStartAction(csaContext =>
             {
                 var obsoleteAttributeType = csaContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemObsoleteAttribute);
-
                 var cultureInfoType = csaContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemGlobalizationCultureInfo);
-                if (cultureInfoType != null)
+
+                if (cultureInfoType == null)
                 {
-                    csaContext.RegisterOperationAction(oaContext =>
+                    return;
+                }
+
+                var excludedSymbols = csaContext.Options.GetExcludedSymbolNamesWithValueOption(Rule, csaContext.Compilation, csaContext.CancellationToken);
```

Comment:
Ah the problem is you are directly invoking this method, instead of invoking the below extension method: https://github.com/dotnet/roslyn-analyzers/blob/3cc0345042a39d23c8b82fd02d0d2dd298dc92cb/src/Utilities/Compiler/Extensions/ISymbolExtensions.cs#L154-L162

This extension method ensures that it skips analysis for a symbol even when any of it's containing symbol is marked for exclusion, which is what the user really desires.

Reply:
Ah! I'll fix this now